crypto/bio/b_addr.c

```diff
@@ -714,11 +719,12 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
             if (hints.ai_flags & AI_ADDRCONFIG) {
                 hints.ai_flags &= ~AI_ADDRCONFIG;
                 hints.ai_flags |= AI_NUMERICHOST;
+                old_ret = gai_ret;
                 goto retry;
             }
```

Comment:
Wouldn't it be simpler if you did `old_ret = gai_ret;` before the `if`?  Then you can use `old_ret` unconditionally below.

Reply:
No, that would overwrite the old_ret with gai_ret from the second pass.

Reply:
D'oh, of course...  Okie, never mind then